src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionImportFunction.cs

```diff
@@ -0,0 +1,177 @@
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Requests;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Client;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using FromBodyAttribute = Microsoft.Azure.Functions.Worker.Http.FromBodyAttribute;
+using ValidationMessages =
+    GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Requests.Validators.ValidationMessages;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+
+public class CompleteNextDataSetVersionImportFunction(
+    ILogger<CompleteNextDataSetVersionImportFunction> logger,
+    PublicDataDbContext publicDataDbContext,
+    IValidator<DataSetVersionProcessRequest> requestValidator)
+{
+    [Function(nameof(CompleteNextDataSetVersionImport))]
+    public async Task<IActionResult> CompleteNextDataSetVersionImport(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = nameof(CompleteNextDataSetVersionImport))] [FromBody]
+        DataSetVersionProcessRequest request,
+        [DurableClient] DurableTaskClient client,
+        CancellationToken cancellationToken)
+    {
+        return await requestValidator
+            .Validate(request, cancellationToken)
+            .OnSuccess(_ => GetNextDataSetVersionInMappingStatus(request, cancellationToken))
+            .OnSuccessDo(_ => GetCompletedDataSetVersionMapping(request, cancellationToken))
+            .OnSuccessCombineWith(nextDataSetVersion =>
+                GetImportInManualMappingStage(request, nextDataSetVersion))
+            .OnSuccess(async versionAndImport =>
+            {
+                var (nextVersion, importToContinue) = versionAndImport;
+
+                importToContinue.InstanceId = Guid.NewGuid();
+                publicDataDbContext.DataSetVersionImports.Update(importToContinue);
+                await publicDataDbContext.SaveChangesAsync(cancellationToken);
+
+                await ProcessCompletionOfNextDataSetVersionImport(
+                    client,
+                    dataSetVersionId: nextVersion.Id,
+                    instanceId: importToContinue.InstanceId,
+                    cancellationToken);
+
+                return new ProcessDataSetVersionResponseViewModel
+                {
+                    DataSetId = nextVersion.DataSetId,
+                    DataSetVersionId = nextVersion.Id,
+                    InstanceId = importToContinue.InstanceId
+                };
+            })
+            .HandleFailuresOr(result => new OkObjectResult(result));
```

Comment:
Not really thought about this until this PR, but I'm wondering if it would be better to put all this logic in a service layer, and treat this like a controller such that it isn't responsible for the business logic. But I realise we haven't been doing this everywhere currently.

Reply:
Suggesting that I absorb this change into 4954 if that's OK?